src/tools/rbd_mirror/PoolWatcher.cc

```diff
@@ -0,0 +1,143 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+#include "common/debug.h"
+#include "common/errno.h"
+#include "PoolWatcher.h"
+
+#define dout_subsys ceph_subsys_rbd_mirror
+#undef dout_prefix
+#define dout_prefix *_dout << "rbd-mirror: "
+
+using std::list;
+using std::map;
+using std::set;
+using std::string;
+using std::unique_ptr;
+using std::vector;
+
+using librados::Rados;
+using librados::IoCtx;
+
+namespace rbd {
+namespace mirror {
+
+namespace {
+
+struct C_RefreshPools : public Context {
+  C_RefreshPools(PoolWatcher *pw) :
+    pw(pw) {}
+  virtual void finish(int r) {
+    pw->refresh_pools();
+  }
+  PoolWatcher *pw;
+};
+
+} // anonymous namespace
+
+PoolWatcher::PoolWatcher(RadosRef cluster, double interval_seconds,
+			 Mutex &lock, Cond &cond) :
+  m_lock(lock),
+  m_refresh_cond(cond),
+  m_cluster(cluster),
+  m_timer(g_ceph_context, m_lock),
+  m_interval(interval_seconds)
+{
+  m_timer.init();
+}
+
+PoolWatcher::~PoolWatcher()
+{
+  m_timer.shutdown();
+}
+
+map<peer_t, set<int64_t> > PoolWatcher::get_peer_configs() const
+{
+  assert(m_lock.is_locked());
+  return m_peer_configs;
+}
+
+void PoolWatcher::refresh_pools()
+{
+  dout(20) << __func__ << dendl;
```

Comment:
as a standalone binary, we can use the global version that relies on g_ceph_context instead of passing cct everywhere like we need to with library version of dout (ldout/lderr)